Core/src/Vertexing/ImpactPointEstimator.cpp

```diff
@@ -6,107 +6,27 @@
-
-  // To understand why deltaR and momDir are not orthogonal, let us look at the
-  // x-y-plane. Since we computed the 3D PCA, the 2D distance between the vertex
-  // and the PCA is not necessarily minimal (see Fig. 4.2 in the reference). As
-  // a consequence, the momentum and the vector connecting the vertex and the
-  // PCA are not orthogonal to each other.
-  Vector3 orthogonalDeltaR = deltaR - (deltaR.dot(momDir)) * momDir;
-
-  // Vector perpendicular to momDir and orthogonalDeltaR
-  Vector3 perpDir = momDir.cross(orthogonalDeltaR);
+namespace Acts {
 
-  // Cartesian coordinate system with:
-  // -) origin at the vertex position
-  // -) z-axis in momentum direction
-  // -) x-axis approximately in direction of the 3D PCA (slight deviations
-  // because it was modified to make if orthogonal to momDir)
-  // -) y-axis is calculated to be orthogonal to x- and z-axis
-  // The transformation is represented by a 4x4 matrix with 0 0 0 1 in the last
-  // row.
-  Transform3 coordinateSystem;
-  // First three columns correspond to coordinate system axes
-  coordinateSystem.matrix().block<3, 1>(0, 0) = orthogonalDeltaR;
-  coordinateSystem.matrix().block<3, 1>(0, 1) = perpDir;
-  coordinateSystem.matrix().block<3, 1>(0, 2) = momDir;
-  // Fourth column corresponds to origin of the coordinate system
-  coordinateSystem.matrix().block<3, 1>(0, 3) = vtxPos;
-
-  // Surface with normal vector in direction of the z axis of coordinateSystem
-  std::shared_ptr<PlaneSurface> planeSurface =
-      Surface::makeShared<PlaneSurface>(coordinateSystem);
-
-  auto intersection =
-      planeSurface
-          ->intersect(gctx, trkParams.position(gctx), trkParams.direction(),
-                      BoundaryCheck(false))
-          .closest();
-
-  // Create propagator options
-  PropagatorOptions<> pOptions(gctx, mctx);
-  pOptions.direction =
-      Direction::fromScalarZeroAsPositive(intersection.pathLength());
-
-  // Propagate to the surface; intersection corresponds to an estimate of the 3D
-  // PCA. If deltaR and momDir were orthogonal the calculation would be exact.
-  auto result =
-      m_cfg.propagator->propagateToSurface(trkParams, *planeSurface, pOptions);
-  if (result.ok()) {
-    return *result;
-  } else {
-    ACTS_ERROR("Error during propagation in estimate3DImpactParameters.");
-    ACTS_DEBUG(
-        "The plane surface to which we tried to propagate has its origin at\n"
-        << vtxPos);
-    return result.error();
-  }
-}
-
-template <unsigned int nDim>
-Acts::Result<double> Acts::ImpactPointEstimator::getVertexCompatibility(
-    const GeometryContext& gctx, const BoundTrackParameters* trkParams,
-    const ActsVector<nDim>& vertexPos) const {
+namespace {
+template <int nDim, typename vector_t>
+Result<double> getVertexCompatibilityImpl(const GeometryContext& gctx,
+                                          const BoundTrackParameters* trkParams,
+                                          vector_t vertexPos) {
```

Comment:
```suggestion
                                          const vector_t& vertexPos) {
```

Reply:
ah I see - yeah no objection. I just thought the template is a bit wide contract but it is source file detail anyways